src/main/scala/mesosphere/marathon/state/Container.scala

```diff
@@ -3,20 +3,31 @@ package mesosphere.marathon.state
 import com.wix.accord.dsl._
 import com.wix.accord._
 import mesosphere.marathon.api.v2.Validation._
-import org.apache.mesos.{ Protos => Mesos }
+
+import org.apache.mesos.{ Protos => mesos }
+
 import scala.collection.immutable.Seq
 
-// TODO: trait Container and specializations?
-// Current implementation with type defaulting to DOCKER and docker to NONE makes no sense
-case class Container(
-    `type`: Mesos.ContainerInfo.Type = Mesos.ContainerInfo.Type.DOCKER,
-    volumes: Seq[Volume] = Nil,
-    docker: Option[Container.Docker] = None) {
+case class Container(`type`: mesos.ContainerInfo.Type,
```

Comment:
As just discussed verbally, I'll now go for a clean Container class and work around this with custom reader/writer as needed.